jboss-cert-ldap-login-module/src/main/java/org/esbtools/auth/jboss/CertLdapLoginModule.java

```diff
@@ -160,15 +152,13 @@ public void initializeRolesProvider() throws Exception {
             LOGGER.debug("Certificate principal:" + certPrincipal);
 
             //first try getting search name from uid in certificate principle (new certificates)
-            String searchName = getLDAPAttribute(certPrincipal, UID);
-            if(StringUtils.isNotBlank(searchName)) {
-                //only try to validate environment if it is a certificate that contains uid
-                validateEnvironment(certPrincipal);
-            } else {
-                // fallback to getting search name from cn in certificate principle (legacy certificates)
-                searchName = getLDAPAttribute(certPrincipal, CN);
+            String searchName = environment.getLDAPAttribute(certPrincipal, UID);
+            if (StringUtils.isBlank(searchName)) {
+                throw new LoginException("A certificate is required.");
```

Comment:
Please update exception message to "A certificate with a UID attribute in the subject name is required."